src/hypergol/tensorflow_model_manager.py

```diff
@@ -121,20 +128,19 @@ def run(self, stepCount, evaluationSteps, tensorboardSteps, metadataSteps, train
             num total steps in schedule
         evaluationSteps: List[int]
             which steps to produce metrics on an evaluation sample
-        tensorboardSteps: List[int]
-            which steps to log taining loss to tensorboard
         metadataSteps: List[int]
             which steps to log metadata to tensorboard
-        trainingSteps: List[int]
-            which steps to train model
         """
-        if trainingSteps is None:
-            trainingSteps = range(stepCount)
-        self.initialize()
-        for k in tqdm(range(stepCount)):
-            if k in trainingSteps:
-                self.train(withLogging=k in tensorboardSteps, withMetadata=k in metadataSteps)
-            if k in evaluationSteps:
-                self.save_model()
-                self.evaluate(withLogging=k in tensorboardSteps, withMetadata=k in metadataSteps)
-        self.save_model()
+        self.start()
+        try:
+            for k in tqdm(range(stepCount)):
+                self.train(withTracing=k in tracingSteps)
```

Comment:
Is this k in tracing steps necessary now? Can’t this just be tracingsteps?

Reply:
withTracing is a boolean, we only save the tracing metadata for certain steps

Reply:
ahh not I see, I think k is just an undescriptive weird variable name